### DISCUSSION TITLE
Possible `runtime execution` workflow being displayed

### DISCUSSION THREAD

**Discussion by Haleshot** [marimo-team/marimo#2485] [Ideas]
Was referring a friend's colab notebook and noticed how Google Colab displays the runtime (underlying methods of libraries that are being executed currently) with a small dot beside the line which is executing. Mostly, this happens in expensive cells (where you are trying out something like model.fit()).

If there are expensive computations; show the methods that are executing and the line in the cell blocks correspondingly. Like a green arrow like sign indicating that the line in the cell block that is executing at that moment and the methods (maybe of a library like the [model.fit](http://model.fit/) function) which are underlying/under the hood operations.

For example:

```python
for rt in range(row):
    for c in range(col):
        for k in range(ch):
            temp = frames[:, rt, c, k] # considering all frames
            temp = temp.reshape(temp.shape[0], 1) # value should be 120 but we won't hardcode it
            gmm.fit(temp)
            means = gmm.means_
            weights = gmm.weights_

            i = np.argmax(weights) # max value which helps in setting the background
            background[rt][c][k] = int(means[i][0])
```

Over here, I had created a `mo.ui.slider` and had used a hyperparameter used in fitting the value. This proved to be expensive as the cell runs for a minimum of 2+ minutes locally.

**Comment by akshayka** [comment]
Whoa very interesting. Can you share a screenshot of what you noticed in Colab? I haven't seen that before

**Reply by Haleshot** [reply]
> Whoa very interesting. Can you share a screenshot of what you noticed in Colab? I haven't seen that before

Will share a demo (gif) which shows what I was talking about above.

**Comment by Haleshot** [comment]
In the demo video attached, you can see how for a brief moment; it shows the line 3 in the cell block executing (with a green dot next to it) and in the footer area of the screen, the underlying methods which are executed is shown for a short period of time.

Could've chosen a better video demo to showcase this in a better fashion; just wanted to give a basic idea. You see it happening in AI/ML related model creation notebooks often.


https://github.com/user-attachments/assets/7b55436a-66c4-4bac-bf91-2d81922a3a1b

**Comment by Haleshot** [comment]
Hope the above demo gave an initial idea/concept of what `runtime execution` means.

**Reply by akshayka** [reply]
Yea, it did. That is pretty cool. They must do some runtime tracing.
